### PR TITLE
Testing 100 coverage with Aggregator

### PR DESCRIPTION
**Pull request by milux**
Trying to reach 100 % coverage (on branches) for Aggregator.

### CODE REVIEW

**Review by oxisto — COMMENTED** [review]
I general, kotlin.test functions such as assertFail should be used instead of junit imports directly - making it easier if we ever go the multiplatform route

File: csaf-schema/src/test/kotlin/io/github/csaf/sbom/ProviderTest.kt [hunk 1]
```diff
@@ -22,51 +22,88 @@ import java.time.OffsetDateTime
...
-                        category = Provider.Category.vendor,
-                        name = "Test Aggregator",
-                        namespace = URI("example.com"),
-                        contact_details = "security@example.com",
-                        issuing_authority = "Very authoritative"
+                    valGen(
+                        Provider::publisher,
+                        Provider.Publisher(
+                            category = Provider.Category.vendor,
+                            name = valGen(Provider.Publisher::name, "Test Aggregator"),
```
Comment (KuechA) [inline csaf-schema/src/test/kotlin/io/github/csaf/sbom/ProviderTest.kt:76]: `"Test Publisher"` for consistency?

File: csaf-schema/src/test/kotlin/io/github/csaf/sbom/PojoTestHelper.kt [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                        }
+                        if (property.returnType.isMarkedNullable) {
+                            validValues += property to null
+                        }
+                        return defaultValue
+                    }
+                }
+            )
+        )
+        println(validValues)
```
Comment (KuechA) [inline csaf-schema/src/test/kotlin/io/github/csaf/sbom/PojoTestHelper.kt:50]: Use a logger instead or `println`?

Comment (milux) [reply csaf-schema/src/test/kotlin/io/github/csaf/sbom/PojoTestHelper.kt:50]: Yes, although in tests, `println()` might be OK. But this is an unintended leftover anyway, so I'll drop that.

**Review by oxisto — APPROVED** [review]

**Pull request merged**